zio-http/src/main/scala/zhttp/socket/SocketProtocol.scala

```diff
@@ -59,53 +57,53 @@ sealed trait SocketProtocol { self =>
     b
   }
 
-}
-
-object SocketProtocol {
-
   /**
    * Close frame to send, when close frame was not send manually.
    */
-  def closeFrame(status: CloseStatus): SocketProtocol = SendCloseFrame(status)
+  def withCloseFrame(status: CloseStatus): SocketProtocol = SocketProtocol.Concat(self, SendCloseFrame(status))
 
   /**
    * Close frame to send, when close frame was not send manually.
    */
-  def closeFrame(code: Int, reason: String): SocketProtocol =
-    SendCloseFrameCode(code, reason)
-
-  /**
-   * Creates an default decoder configuration.
-   */
-  def default: SocketProtocol = Default
+  def withCloseFrame(code: Int, reason: String): SocketProtocol =
+    SocketProtocol.Concat(self, SendCloseFrameCode(code, reason))
 
   /**
    * Close the connection if it was not closed by the client after timeout
    * specified
    */
-  def forceCloseTimeout(duration: Duration): SocketProtocol =
-    ForceCloseTimeoutMillis(duration)
+  def withForceCloseTimeout(duration: Duration): SocketProtocol =
+    SocketProtocol.Concat(self, ForceCloseTimeoutMillis(duration))
 
   /**
    * Close frames should be forwarded
    */
-  def forwardCloseFrames: SocketProtocol = ForwardCloseFrames
+  def withForwardCloseFrames: SocketProtocol = SocketProtocol.Concat(self, ForwardCloseFrames)
```

Comment:
Once enabled there should be a way to disable also. Otherwise the builder pattern will be a one way function.

We can pass a boolean to solve this and name it as `forwardCloseFrames(false)`.

This is applicable to all the other operators also.

Reply:
By default all are disabled, we are enabling the ones that we are need it as, once protocol defined,  and passed to socket app, it is not modified anymore. It is like one time thing created before the app it is started.... that's the reason why I did not added a way to disable them..